### ISSUE TITLE
How to use TrilioVault in PSS-enforced/hardened clusters

### ISSUE THREAD

**Issue by josedev-union** [triliovault-k8s-issues/triliovault-k8s-issues#33]
## Issue Description
Following an alpha release in Kubernetes v1.22, Pod Security Admission became available by default in Kubernetes v1.23, as a beta. From version 1.25 onwards, Pod Security Admission is generally available.
Following the K8s CIS benchmarks, we set `restricted` PSS(Pod Security Standard) as the cluster-wide default. https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted
All backup and restore jobs run inside the application(to backup) namespaces which are enforced by the restricted PSS. But pods are not scheduling because those jobs are all violating the PS standard. Even preflight check is failing because of the same violation.
```
ERRO[0001] ❌ Preflight check for pod capability failed :: pods "pod-capability-0-sdibok" is forbidden: violates PodSecurity "baseline:latest": non-default capabilities (container "busybox" must not include "SYS_ADMIN" in securityContext.capabilities.add), privileged (container "busybox" must not set securityContext.privileged=true) 
```
We can run the preflight check in a specific namespace which has the `privileged` enforcement. But we cannot set `privileged` standard for any namespaces where applications are running.

### Preconditions
```
1. k8s flavour: Any
2. k8s version: 1.25.x>=
3. storage: Any
4. CSI version: Any
5. k8s TrilioVault version: 2.x, 3.x
...
```

### Steps to reproduce

```
1. Create Target named nfs-target
2. Created BackupPlan named mysql-backup-plan
3. Created Backup with name mysql-backup
...
```

### Actual and Expected result
Actual: No pod is scheduling for trilio actions(back, restore, etc.)
Expected result: Backup, restore jobs should follow the PSS. If any jobs require a specific permission, it should be documented so users can allow it in the application namespaces.

### Support Log Bundle
No need to attach logs

**Comment by madhurtrilio** [comment]
Hi @josedev-union thanks for reporting your issue. We understand the problem and we are working on resolving this. The solution that we are coming up is that all our backup/restore jobs will run in the namespace where TVK will be installed. With that you will not have to give permissions to the namespaces where the applications are running. This work is in progress and we are targtting it for release version 3.2.0. It should be GA by July end

**Comment by josedev-union** [comment]
> Hi @josedev-union thanks for reporting your issue. We understand the problem and we are working on resolving this. The solution that we are coming up is that all our backup/restore jobs will run in the namespace where TVK will be installed. With that you will not have to give permissions to the namespaces where the applications are running. This work is in progress and we are targtting it for release version 3.2.0. It should be GA by July end

yeah, that is a good solution. Also preflight should be executed in the trilio namespace but it could be tricky because it is a kubectl plugin which uses the namespace specified in the cli or kubecontext.
Btw, thanks for your answer 👍
